### PR TITLE
Feat: Save model and visualize performance, remove unused code and add invoke commands.

### PR DESCRIPTION
**Pull request by s113117**
- Remove unused code.
- Update training script to save the model (in /models).
- Add YAML config for model parameters for tasks.
- Add invoke tasks for training and visualization.
 ```bash
 invoke train
   ```
 ```bash
invoke visualize
   ```
- Add visualization script to generate training loss and TSNE plots.
- Add dependency for Lightning.
- Update tests to validate dataset.

train_losses

tsne

### CODE REVIEW

**Comment by radugrecu97** [comment]
I setup invoke with the LightningCLI trainer on my branch.

I understood that the yaml config is autogenerated for each run from the passed arguments with LightningCLI.

You can of course specify the config file as an argument.


Do we specify which file to run on google cloud?

**Comment by s113117** [comment]
"I setup invoke with the LightningCLI trainer on my branch."

- Oh, yeah. That's right. I just extended it to take more arguments

"I understood that the yaml config is autogenerated for each run from the passed arguments with LightningCLI."

- The added yaml file is only for the invoke commands (as _train_ and _visualize_ take the same arguments). It is unrelated to the rest.

"Do we specify which file to run on google cloud?"

- Yep, in train.dockerfile. We build the image using Google Build and run it as a custom job on Vertex AI.
